### DISCUSSION TITLE
MO rotation and use MO as input

### DISCUSSION THREAD

**Discussion by Emilioerc** [pyscf/pyscf#2711] [Q&A]
Dear all,

I am trying to get the MO of one particular geometry and use them to obtain the energy for another geometry, so that all my geometries are constructed in the same MO basis or at least in a quite similar basis.

I tried looking at the overlap in that way: overlap = np.dot(myhf.mo_coeff.T, myhf1.mo_coeff). After that, I try to change one of the mo_coeff by applying a rotation matrix. But then, how can I use my personalized MO as input for the calcultions, instead of as a guess? Another query is that wether this rotation is alredy implemented, in order to compare

Thank you in advance!

**Comment by fishjojo** [comment]
You can construct a density matrix from your MOs, then the SCF calculation can use this density matrix as the initial guess:
```
mf = scf.RHF(mol)
mf.kernel(dm0=your_dm)
```
For MO projections, see `scf.addons.project_mo_nr2nr`

**Comment by Emilioerc** [comment]
> You can construct a density matrix from your MOs, then the SCF calculation can use this density matrix as the initial guess:
> 
> ```
> mf = scf.RHF(mol)
> mf.kernel(dm0=your_dm)
> ```
> 
> For MO projections, see `scf.addons.project_mo_nr2nr`

Thank you very much for your answer. I have tried to do it and it can take it as a guess but finally the orbital basis is changes with repect to the reference geometry. I would like to construct a 2D PES in which the  molecular orbital basis is the same in all the surface, but I am not able to achieve it. Can you give me any advise, is an easy problem to understand, but not so easy in the practice.

**Comment by verena-neufeld** [comment]
Maybe you can try setting `mf.max_cycle = 0` before `mf.kernel(dm0=your_dm)`?

**Comment by verena-neufeld** [comment]
Hi @Emilioerc , has your question been answered?

**Comment by Emilioerc** [comment]
> Hi @Emilioerc , has your question been answered?

No, it did not work. I just wanted to diabatize the Hamiltonian, but I was not able to do it.

**Comment by sunqm** [comment]
Using the `energy_tot` method
```
# define mol1 for one geometry
mf1 = mol1.RHF().run()
e1 = mf1.e_tot
mf1.make_rdm1()

# define mol2 for another geometry
mf2 = mol2.RHF()
nelec2 = np.einsum('ij,ji', mf2.get_ovlp(), dm1)
e2 = mf2.energy_tot(mol2.nelectron/nelec2 * dm1)
```

Please note the MOs of the first geometry are not strictly orthonormalized for the second geometry. This may lead to wrong number of electrons and unphysical total energy. Therefore, the density matrix is scaled to produce right number of electrons.
